test/index.js

```diff
@@ -1,701 +1,813 @@
-    const sendDiagnostics = /** @type import('sinon').SinonStub */ (
-      connection.sendDiagnostics
-    )
-    sendDiagnostics.callsFake(resolve)
-  })
-  const onDidChangeContent = spy()
-  Object.defineProperty(documents, 'onDidChangeContent', {
-    value: onDidChangeContent
-  })
-
-  configureUnifiedLanguageServer(connection, documents, {
-    plugins: [[pluginName, pluginOptions]],
-    processor
-  })
-
-  onDidChangeContent.firstCall.firstArg({
-    document: TextDocument.create(uri, 'text', 0, text)
-  })
-
-  return diagnosticsPromise
-}
 
-test('onInitialize', (t) => {
-  const connection = createMockConnection()
-  const documents = new TextDocuments(TextDocument)
+import * as exports from 'unified-language-server'
 
-  configureUnifiedLanguageServer(connection, documents, {})
+const sleep = promisify(setTimeout)
 
-  const initialize = /** @type import('sinon').SinonSpy */ (
-    connection.onInitialize
-  ).firstCall.firstArg
-  const result = initialize()
+const delay = process.platform === 'win32' ? 600 : 300
+const timeout = 10_000
 
-  t.deepEquals(result, {
-    capabilities: {
-      textDocumentSync: TextDocumentSyncKind.Full,
-      documentFormattingProvider: true,
-      codeActionProvider: {
-        codeActionKinds: [CodeActionKind.QuickFix],
-        resolveProvider: true
-      }
-    }
-  })
+test('exports', (t) => {
+  t.equal(typeof exports.createUnifiedLanguageServer, 'function')
 
   t.end()
 })
 
-test('Custom processor', async (t) => {
-  const uri = String(pathToFileURL('test.md'))
-  const diagnostics = await getDiagnostic(
-    uri,
-    'test',
-    undefined,
-    unified().use(
-      /** @type {Plugin} */ (
-        () => (ast, file) => {
-          file.message('custom processor')
-        }
-      )
-    )
+test('`initialize`', async (t) => {
+  await fs.writeFile(
+    new URL('lsp.js', import.meta.url),
+    `import {createUnifiedLanguageServer} from '../index.js'
+    createUnifiedLanguageServer({plugins: ['remark-parse', 'remark-stringify']})`
   )
 
-  t.deepEquals(diagnostics, {
-    uri,
-    version: 0,
-    diagnostics: [
-      {
-        range: {start: {line: 0, character: 0}, end: {line: 0, character: 0}},
-        message: 'custom processor',
-        severity: DiagnosticSeverity.Warning
-      }
-    ]
-  })
-
-  t.end()
-})
+  const stdin = new PassThrough()
 
-test('onDocumentFormatting different', async (t) => {
-  const connection = createMockConnection()
-  const documents = new TextDocuments(TextDocument)
-  const uri = String(pathToFileURL('test.md'))
-  const get = stub(documents, 'get').returns(
-    TextDocument.create(uri, 'markdown', 0, '#   Hello world!')
-  )
-
-  configureUnifiedLanguageServer(connection, documents, {
-    plugins: ['remark-parse', 'remark-stringify']
+  const promise = execa('node', ['lsp.js', '--stdio'], {
+    cwd: fileURLToPath(new URL('.', import.meta.url)),
+    input: stdin,
+    timeout
   })
```

Comment:
Maybe tests can be solved [`vscode-jsonrpc`](https://github.com/microsoft/vscode-languageserver-node/tree/main/jsonrpc) and without sleep.

```js
import cp from 'node:child_process'

import {
  createMessageConnection,
  StreamMessageReader,
  StreamMessageWriter
} as rpc from 'vscode-jsonrpc/node'

test('', async () => {
  const diagnostics = await new Promise((resolve, reject) => {
    let childProcess = cp.spawn(...)

    const connection = createMessageConnection(
      new StreamMessageReader(childProcess.stdout),
      new StreamMessageWriter(childProcess.stdin)
    )

    connection.onNotification(() => {
      // XXX implement this
      connection.end()
      childProcess.kill()
      resolve()
    })
  })
})
```

Reply:
I dunno. It’s a library without documentation. I have no clue how to use it.

Reply:
I’ve tried for 30 minutes. I have no clue how to use it (and I can’t imagine other people having fun with a library without docs either). LSP is well documented and works well.